pin docs incorrectly claims that users can pin from git url

# Problem

`opam pin` from git url does not work exactly as described

### Works

The below combo of `opam pin -n ...` + `opam install <pkg>` take:

```
❯ opam pin add -y -n git+http://github.com/EduardoRFS/ocamlearlybird#2e37822
[ocamlearlybird] synchronised from git+http://github.com/EduardoRFS/ocamlearlybird#2e37822
[earlybird.0.1.4] synchronised from git+http://github.com/EduardoRFS/ocamlearlybird#2e37822
earlybird is now pinned to git+http://github.com/EduardoRFS/ocamlearlybird#2e37822 (version 0.1.4)

❯ opam install earlybird
# ok
```

### Fails

However, the below fail, seemingly in contrast to https://opam.ocaml.org/doc/Usage.html#opam-pin . 

`opam pin add <pkg> <url>#<commit>`:
```
❯ opam pin add earlybird http://github.com/EduardoRFS/ocamlearlybird#2e37822
[earlybird.0.1.4] downloaded from http://github.com/EduardoRFS/ocamlearlybird#2e37822
[ERROR] Error getting source from http://github.com/EduardoRFS/ocamlearlybird#2e37822:
          - Unknown archive type: /private/var/folders/nx/8ddl3q8d3cq8r89p8m9rt8gc0000gn/T/opam-13735-05296d/ocamlearlybird
```

`opam pin add <pkg> <url-branch>`
```
❯ opam pin https://github.com/EduardoRFS/ocamlearlybird/tree/fold-values-workaround-4.09
[fold-values-workaround-4] downloaded from https://github.com/EduardoRFS/ocamlearlybird/tree/fold-values-workaround-4.09
[ERROR] Could not retrieve Unknown archive type:
        /private/var/folders/nx/8ddl3q8d3cq8r89p8m9rt8gc0000gn/T/opam-13246-9b8556/fold-values-workaround-4.09
```

Same as above with `opam pin add <pkg> <url>#<branch>`

`opam pin <url-branch>.git`
```
❯ opam pin https://github.com/EduardoRFS/ocamlearlybird/tree/fold-values-workaround-4.09.git
[ERROR] Could not synchronize
        /private/var/folders/nx/8ddl3q8d3cq8r89p8m9rt8gc0000gn/T/opam-pin-cache-13305-5a61c5/2b6cb074ff0fb73e8f0eead2bc854d562a774953287af766c449728ef69332c189d51381e475b4f6ebda042e8b5775ac9112c2a9a13034621de7f68d10d47f1c
        from "git+https://github.com/EduardoRFS/ocamlearlybird/tree/fold-values-workaround-4.09.git":
        "/usr/local/bin/git fetch -q" exited with code 128 "fatal: repository
        'https://github.com/EduardoRFS/ocamlearlybird/tree/fold-values-workaround-4.09.git/' not found"
[ERROR] Could not retrieve git+https://github.com/EduardoRFS/ocamlearlybird/tree/fold-values-workaround-4.09.git
```

--- a/doc/pages/Usage.md
+++ b/doc/pages/Usage.md
@@ -128,11 +128,12 @@ The syntax is
 opam pin add <package name> <target>
 ```
 
-Where `<target>` may be a version, but also a local path, the URL of an archive,
-or even a git, mercurial or darcs URL. The package will be kept up-to-date with
-its origin on `opam update` and when explicitly mentioned in a command, so that
-you can simply run `opam upgrade <package name>` to re-compile it from its
-upstream. If the upstream includes opam metadata, that will be used as well.
+Where `<target>` may be a version, but also an [URL](Manual.html#URLs) of a
+local path, an archive, or even a git, mercurial or darcs repository. The
+package will be kept up-to-date with its origin on `opam update` and when
+explicitly mentioned in a command, so that you can simply run `opam upgrade
+<package name>` to re-compile it from its upstream. If the upstream includes
+opam metadata, that will be used as well.
 
 ```
 opam pin add camlpdf 1.7                                      # version pin

--- a/master_changes.md
+++ b/master_changes.md
@@ -114,4 +114,4 @@ New option/command/subcommand are prefixed with ◈.
 
 ## Doc
   * Update ùaintainer name [#4456 @nbraud]
-  *
+  * Specify url syntaxe in usage/opam pin [#4460 @rjbou - fix #4459]

